### PR TITLE
SDCICD-437: Support separate addon cleanup harnesses

### PR DESCRIPTION
**Pull request by jeefy**
This adds the `ADDON_CLEANUP_HARNESSES` config option and updates the docs to support it.

### CODE REVIEW

File: pkg/common/config/config.go [hunk 1]
```diff
@@ -250,12 +250,17 @@ var Addons = struct {
...
-	IDs:           "ids",
-	TestHarnesses: "testHarnesses",
-	TestUser:      "addons.testUser",
-	RunCleanup:    "addons.runCleanup",
+	IDsAtCreation:    "addons.idsAtCreation",
+	IDs:              "ids",
+	TestHarnesses:    "testHarnesses",
+	TestUser:         "addons.testUser",
+	RunCleanup:       "addons.runCleanup",
+	CleanupHarnesses: "cleanupHarnesses",
```
Comment (meowfaceman) [inline pkg/common/config/config.go:263]: Huh, odd. I have a nit to expand all of these to addons.*, so:

```
addons.ids
addons.testHarnesses
addons.cleanupHarnesses
```

I must have forgotten this when I set this up. I realize that ids/testHarnesses isn't part of your change, but mind fixing them while we're here?

File: pkg/e2e/e2e.go [hunk 2]
```diff
@@ -503,8 +503,16 @@ func cleanupAfterE2E(h *helper.H) (errors []error) {
...
 	log.Printf("Addon cleanup: %v", viper.GetBool(config.Addons.RunCleanup))
 	if viper.GetBool(config.Addons.RunCleanup) {
+		// By default, use the existing test harnesses for cleanup
+		harnesses := strings.Split(viper.GetString(config.Addons.TestHarnesses), ",")
+
+		// Check if cleanup harnesses exist and if so, use those instead
+		cleanupHarnesses := viper.GetString(config.Addons.CleanupHarnesses)
+		if len(cleanupHarnesses) > 0 {
+			harnesses = strings.Split(viper.GetString(config.Addons.CleanupHarnesses), ",")
+		}
```
Comment (meowfaceman) [inline pkg/e2e/e2e.go:513]: Think we need an else here.

Comment (jeefy) [reply pkg/e2e/e2e.go:513]: I don't think we do. I default the cleanup harnesses to be the previously-set test harness, and overwrite `harnesses` only if it's present/valid.

Comment (meowfaceman) [reply pkg/e2e/e2e.go:513]: Oh, I'm sorry. I overlooked the assign -- yes, please disregard.

File: docs/Addons.md [hunk 3]
```diff
@@ -32,6 +30,17 @@ env:
...
 
+### Addon Cleanup
+
+If your addon test creates or affects anything outside of the OSD cluster lifecycle, a separate cleanup action is required. If `ADDON_RUN_CLEANUP` is set to `true`, OSDe2e will run the test harness a **second time** passing the argument `cleanup` to the container.
+
+There may be a case where a separate cleanup container/harness is required. That may be configured using the `ADDON_CLEANUP_HARNESSES` config option. It is formatted in the same way as `ADDON_TEST_HARNESSES`. This however, may cause some confusion as to what is run when:
+
+`ADDON_RUN_CLEANUP` is true, and `ADDON_CLEANUP_HARNESSES` is not set, OSDe2e will only run `ADDON_TEST_HARNESSES` again, passing the `cleanup` argument.
+
+`ADDON_RUN_CLEANUP` is true, and `ADDON_CLEANUP_HARNESSES` is set, OSDe2e will only run the `ADDON_CLEANUP_HARNESSES`, passing the `cleanup` argument.
```
Comment (Boomatang) [inline docs/Addons.md:41]: As the `ADDON_CLEANUP_HARNESSES` can only be run when the `ADDON_RUN_CLEANUP` flag is set, I don't see the need to pass in the `cleanup` argument.

Comment (jeefy) [reply docs/Addons.md:41]: That's a fair point. :) I'll update the docs and logic quick.

File: pkg/e2e/e2e.go (hunk 2 above)
Comment (Boomatang) [reply pkg/e2e/e2e.go:513]: On the if/else statement, I feel h.RunAddonTests() should possible be move in to so not to have to pass the "cleanup" flag to the clean up harnesses

File: pkg/e2e/e2e.go [hunk 4]
```diff
@@ -503,8 +503,18 @@ func cleanupAfterE2E(h *helper.H) (errors []error) {
...
 	log.Printf("Addon cleanup: %v", viper.GetBool(config.Addons.RunCleanup))
 	if viper.GetBool(config.Addons.RunCleanup) {
+		// By default, use the existing test harnesses for cleanup
+		harnesses := strings.Split(viper.GetString(config.Addons.TestHarnesses), ",")
+		arguments := []string{"cleanup"}
+
+		// Check if cleanup harnesses exist and if so, use those instead
+		cleanupHarnesses := viper.GetString(config.Addons.CleanupHarnesses)
+		if len(cleanupHarnesses) > 0 {
+			harnesses = strings.Split(viper.GetString(config.Addons.CleanupHarnesses), ",")
```
Comment (meowfaceman) [inline pkg/e2e/e2e.go:513]: nit: One last nit I missed -- just reuse `cleanupHarnesses` here instead of doing the second `viper.GetString`.

Comment (jeefy) [reply pkg/e2e/e2e.go:513]: oh duh

**Pull request merged**